File: Submissions/594_Longest_Harmonious_Subsequence.cpp
class Solution {
public:
    int findLHS(vector<int>& nums) {
        int result = 0;
        map<int, int> frequencys;
        
        for(auto& num : nums){
            ++frequencys[num];
        }
        
        for(auto& frequency : frequencys){
            auto found = frequencys.find(frequency.first+1);
            if(found != frequencys.end()) {
                result = max(result, frequency.second + found->second);
            }
        }
        return result;
    }
};

File: Kotlin/594_Longest_Harmonious_Subsequence.kt
class Solution {
    fun findLHS(nums: IntArray): Int {
        var result = 0
        val frequency = mutableMapOf<Int, Int>()
        nums.forEach{
            frequency[it] = frequency.getOrDefault(it, 0) + 1
        }
        
        frequency.forEach{
            if(frequency.containsKey(it.key + 1)) {
                result = maxOf(result, it.value + frequency[it.key + 1]!!)
            }
        }

        return result
    }
}